config/v1/types_ingress.go

```diff
@@ -152,8 +153,53 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is present only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer. For Example: Setting AWS EIPs https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations assign Elastic IP addresses to the Network Load Balancer.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
```

Comment:
As a cluster administrator who is configuring `eipAllocations`, how do I know the number of subnets that are used for the load balancer?  Does this field determine which subnets are used?

Reply:
Actually I am thinking of working on the installer part in the next release. The installer will need to check the number of availability zones.

Reply:
For validation for BYOB subnets: We will count if the number of cluster subnets provided by the user in the install-config.yaml were equal to the number of the eip allocations provided in the install-config.yaml. We will need to add validations in [pkg/asset/installconfig/aws/validation.go](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/validation.go#L186)
For validation for Installer Created Subnets: The number of EIPs must match number of AZs in region.

Reply:
The godoc doesn't necessarily need to explain the implementation details of how the installer and operator use the field or value; rather, the godoc needs to explain what the field or value means.  

Is the following true?  "The number of entries for eipAllocations must be equal to the number of availability zones in which the cluster has nodes."  I expect some straightforward statement like that  in the API godoc.

Reply:
Is the following true?  Yes from understanding. @gcs278 WDYT ?